config/opal_dep_version.m4

```diff
@@ -0,0 +1,51 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2022      IBM Corporation.  All rights reserved.
+dnl
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+# OPAL_GET_DEP_VERSION(version_file, variable_prefix)
+# -----------------------------------------------
+# Parse version_file for dependent version information, setting
+# the following shell variables:
+# OMPI_$2_MIN_VERSION
+# OMPI_$2_NUMERIC_MIN_VERSION
+#
+AC_DEFUN([OPAL_GET_DEP_VERSION],[
```

Comment:
I would make variable_prefix be a M4 literal (which is the usage in this patch).  Then you can move the upper/lower logic to `m4_toupper` and `m4_tolower` and have considerably simpler macro.  You could also almost certainly move the parsing logic to m4 as well, which would make the chance of errors move to Autoconf time, which is way better.

Also, I have objections to the AC_MSG_NOTICE; there's no reason customers would care.

Reply:
Good point - I updated the PR. This does simplify it quite a bit.

Regarding the parsing of the string - I am not sure how to do that in m4 - do we have an example in our configure logic somewhere?

Reply:
It seems like we can do a bunch of this during autoconf time, instead of configure time, similar to parsing the major/minor/release versions, but I guess this is good enough for now.